### DISCUSSION TITLE
Review of BIP 330 : Transaction announcement reconciliation

### DISCUSSION THREAD

**Discussion by Prabhat1308** [Bitshala/BitcoinCore-PR-Review-Club#67] [PR Reviews]
# Session Details
Date : 25-04-2024
Time: IST 20:00 (UTC 14:30)
Link : [BIP 330](https://github.com/bitcoin/bips/blob/master/bip-0330.mediawiki)
BIP : 330 
Difficulty : Medium

## Notes
Currently everyone announcement of transaction takes up 32 byte for each transaction ID atleast once in every P2P connection . This is very inefficient and takes up lot of Bandwidth . This BIP introduces another way for declaration of transactions with the newly introduced protocol Erlay.

## Questions

- What is bandwidth and why is this new protocol important ?
- What exactly is Erlay ? is it backward compatible ?
- What are short transaction IDs ?
- Reconciliation scheme is asymmetric . If we have an outbound peer we initiate the reconciliation. Why not a symmetric design ?  
- The private nodes almost always lag behind in their mempool because private nodes don’t have incoming connections  ?
- With the introduction of erlay , the nodes need to store extra data inform of sketches and need to support its operations. 
- why a reconciliation-based approach scales better than flooding ? 
- frequency of reconciliation fixed at 1 every 2 seconds ? Why fixed and not dynamic? 
- Analyse the effectiveness of this protocol

## Learnings 
Bandwith
Erlay
New p2p messages 

## Support Material
[Erlay](https://arxiv.org/pdf/1905.10518.pdf)

**Comment by yuvicc** [comment]
The project implementation to Bitcoin-Core can be tracked here!

https://github.com/bitcoin/bitcoin/issues/28646
